api/pkg/client/mlflow/mlflow.go

```diff
@@ -0,0 +1,188 @@
+	Value string `json:"value"`
+}
+type DataRun struct {
+	Tags []TagRun `json:"tags"`
+}
+
+type InfoRun struct {
+	RunId          string `json:"run_id"`
+	ExperimentId   string `json:"experiment_id"`
+	UserId         string `json:"user_id"`
+	LifecycleStage string `json:"lifecycle_stage"`
+	ArtifactURI    string `json:"artifact_uri"`
+}
+type RunResponse struct {
+	Info InfoRun `json:"info"`
+	Data DataRun `json:"data"`
+}
+type SearchRunsResponse struct {
+	RunsData []RunResponse `json:"runs"`
+}
+type SearchRunResponse struct {
+	RunData RunResponse `json:"run"`
+}
+
+func NewMlflowClient(httpClient *http.Client, config Config) *mlflowClient {
+	return &mlflowClient{
+		Client: httpClient,
+		Config: config,
+	}
+}
+
+func (mfc *mlflowClient) httpCall(method string, url string, headers map[string]string, body []byte, response interface{}) error {
+	req, err := http.NewRequest(method, url, bytes.NewBuffer(body))
+	if err != nil {
+		return err
+	}
+
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+
+	resp, err := mfc.Client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if !(resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices) {
+		// Convert response body to Error Message struct
+		var errMessage DeleteExperimentErrorResponse
+		if err := json.NewDecoder(resp.Body).Decode(&errMessage); err != nil {
+			return err
+		}
+		return fmt.Errorf(errMessage.Message)
+	}
+
+	if response != nil {
+		if err := json.NewDecoder(resp.Body).Decode(response); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (mfc *mlflowClient) SearchRunForExperiment(idExperiment string) (SearchRunsResponse, error) {
+	// Search related runs for an experiment id
+	var responseObject SearchRunsResponse
+
+	searchRunURL := fmt.Sprintf("%s/api/2.0/mlflow/runs/search", mfc.Config.TrackingURL)
+
+	headers := map[string]string{
+		"Content-Type": "application/json",
+	}
+
+	input := SearchRunRequest{ExperimentId: []string{idExperiment}}
+	jsonInput, err := json.Marshal(input)
+	if err != nil {
+		return responseObject, err
+	}
+
+	err = mfc.httpCall("POST", searchRunURL, headers, jsonInput, &responseObject)
+	if err != nil {
+		return responseObject, err
+	}
+
+	return responseObject, nil
+}
+
+func (mfc *mlflowClient) SearchRunData(idRun string) (SearchRunResponse, error) {
+	// Creating Output Format for Run Detail
+	var runResponse SearchRunResponse
+	getRunURL := fmt.Sprintf("%s/api/2.0/mlflow/runs/get?run_id=%s", mfc.Config.TrackingURL, idRun)
+
+	err := mfc.httpCall("GET", getRunURL, nil, nil, &runResponse)
+	if err != nil {
+		return runResponse, err
+	}
+	return runResponse, nil
+}
+
+func (mfc *mlflowClient) DeleteExperiment(idExperiment string) error {
+	// Creating Input Format for Delete experiment
+	input := DeleteExperimentRequest{ExperimentId: idExperiment}
+	// HIT Delete Experiment API
+	delExpURL := fmt.Sprintf("%s/api/2.0/mlflow/experiments/delete", mfc.Config.TrackingURL)
```

Comment:
Since we only do a soft delete of the experiment here, it will render the experiment name unusable in the future, right? My understanding was, we decided to let the DeleteExperiment function only issue a soft delete call to all its runs instead, and leave the experiment intact. CMIIW @deadlycoconuts @Zenovore

Reply:
> My understanding was, we decided to let the DeleteExperiment function only issue a soft delete call to all its runs instead, and leave the experiment intact.

Yes, but I think what makes the DeleteExperiment function truly a 'soft' delete and not a 'hard' one is that we're not removing that experiment from the mlflow-db, and not only because it 'only issue[s] a soft delete call to all its runs'.

Reply:
> but I think what makes the DeleteExperiment function truly a 'soft' delete and not a 'hard' one is that we're not removing that experiment from the mlflow-db

It's not about "removing the experiment", I mean we shouldn't even soft delete the Experiment record (because it will cause the name conflict later on). So, Delete Experiment doesn't modify the experiment record in any way. Hope that makes sense.

Reply:
Ahh, ok got it, okay I must've interpreted the decision slightly differently then (I thought we were okay with leaving the name conflict because we were going to clean up the mlflow db too subsequently). Thanks for clearing it up!

Reply:
Yes, in the long term we can aim to hard delete the experiment properly, but in the short term probably better to leave the experiment record untouched to prevent the name conflict.